### PR TITLE
Add typescript

### PR DESCRIPTION
**Pull request by Tammo-Feldmann**
This touches a lot of the code base so we'd probably want to merge this asap. Happy to discuss if you have any questions.

### CODE REVIEW

File: app/catalog.json [hunk 1]
```diff
@@ -9,24 +9,24 @@
       "rel": "child",
       "type": "application/json",
       "title": "Microsoft Planetary Computer",
-      "heystac:id": "microsoft-pc",
+      "heystacId": "microsoft-pc"
```
Comment (gadomski) [inline app/catalog.json:12]: The `ident:attribute` syntax is used for [STAC extensions](https://stac-extensions.github.io/), so I'd like to keep it.

File: app/catalogs/[id]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import { notFound } from "next/navigation";
+
+interface Catalog {
+  type: string;
+  stac_version: string;
+  id: string;
+  description: string;
+  links: Link[];
+}
+
```
Comment (gadomski) [inline app/catalogs/[id]/page.tsx:13]: This is derived directly from the STAC specification, so at the least we should move the interface up to `app/catalogs` — or is there a more common place to store interfaces for a next.js app?

File: app/catalogs/[id]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+type LinkIdObject = { id: string };
+
+interface Link {
+  href: string;
+  rel: string;
+  type: string;
+  title: string;
+  heystacId: string;
+}
```
Comment (gadomski) [inline app/catalogs/[id]/page.tsx:22]: Same as `Catalog`, this is an interface we'll re-use elsewhere so we should move it higher in `app/`.

File: app/catalogs/[id]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+interface Catalog {
+  type: string;
+  stac_version: string;
+  id: string;
+  description: string;
+  links: Link[];
+}
+
+type LinkIdObject = { id: string };
```
Comment (gadomski) [inline app/catalogs/[id]/page.tsx:14]:
```suggestion
type Params = { id: string };
```

**Review by gadomski — CHANGES_REQUESTED** [review]
I did a quick search-around for existing typescript definitions for the STAC spec, found https://github.com/blacha/stac-ts which seems like the best fit (it's the only hit for typescript on https://stacindex.org/ecosystem?language=TypeScript). Could/should we lean on that rather than defining our own?

File: app/catalog.json (hunk 1 above)
Comment (Tammo-Feldmann) [reply app/catalog.json]: Makes sense. Thanks for the background.

**Comment by Tammo-Feldmann** [comment]
I made the changes that you suggested. Good call on those.

**Comment by Tammo-Feldmann** [comment]
I like https://github.com/blacha/stac-ts as another improvement. Haven't made that change yet, though. Need to call it for tonight.

**Comment by gadomski** [comment]
> I like https://github.com/blacha/stac-ts as another improvement.

Word, make #12 to track.

File: app/catalogs/[id]/page.tsx [hunk 5]
```diff
@@ -1,19 +1,23 @@
 import CatalogPage from "../../components/pages/catalog";
-import Catalog from "../../catalog";
+import Root from "../../catalog.json";
```
Comment (gadomski) [inline app/catalogs/[id]/page.tsx:2]: This feels more clear to me (rather than aliasing the `Catalog` type below.

**Review by gadomski — APPROVED** [review]

**Comment by gadomski** [comment]
@Tammo-Feldmann are there any checks we should add to our CI? Or will the build system take care of that for us now?

**Comment by Tammo-Feldmann** [comment]
> are there any checks we should add to our CI?
The type check at compile time is a good step in the right direction. I mean, there is a bunch of useful stuff out there but I don't know where you want to take this app. The obvious starting points would be Jest for unit tests and either Cypress or Playwright for e2e. A year ago I would have recommended Playwright pretty unequivocally and I still would if you want to test multiple languages with the same syntax/framework. Otherwise, Cypress now has component testing that obviates React Testing Library. That is a pretty nice thing to have together in the same framework and Cypress has made their UX really nice in the last few releases.

**Comment by gadomski** [comment]
Excellent, glad we're good on the typescript side of things. I'll copy this recommendation to https://github.com/gadomski/heystac/issues/6

**Pull request merged**